lib/BackbeatConsumer.js

```diff
@@ -535,6 +540,72 @@ class BackbeatConsumer extends EventEmitter {
         return undefined;
     }
 
+    /**
+     * Pauses consumption of assigned partitions
+     * @returns {undefined}
+     */
+    _pauseAssignments() {
+        const assignments = this._consumer?.assignments();
+        // removing offset from assignments
+        const topicPartitions = assignments?.map(({ topic, partition }) => ({ topic, partition }));
+        if (!topicPartitions || topicPartitions.length === 0) {
+            return;
+        }
+        this._consumer.pause(topicPartitions);
+        topicPartitions.forEach(({ topic, partition }) => {
+            this._pausedPartitions.set(`${topic}:${partition}`, { topic, partition });
```

Comment:
why do we keep track of the "paused" partitions in a separate variable: can't we just simply retrieve the (current) assingments every time?

Reply:
Yeah it doesn't make sense to store it separately since we resume before unassigning. The current assignments  should always contain the full list of paused partitions.